Storybook

Jhon-Crow opened the pull request.

jhamish committed:
начинаю сторибук
выдаёт Invalid or unexpected token
       SyntaxError: Invalid or unexpected token

jhamish committed:
.storybook

jhamish committed:
удалил лишнюю папку storybook
button-component.styles.js не редактируется, подозреваю ошибка из за него

jhamish committed:
сверился с образцом, моя кнопка в сторибуке не появилась

jhamish committed:
работает

jhamish committed:
добавил header-nav-item - не отображается

Yashiandr commented on .storybook/preview.js:
```diff
@@ -0,0 +1,20 @@
+
+[
+	ButtonComponent,
+].map((component) => customElements.define(component.name, component));
```
не указал HeaderNavItemComponent здесь, вот он у тебя и не прогружает этот компонент для HeaderNavItem в сторибуке.

Yashiandr commented on src/components/header-nav-item/header-nav-item-component.styles.js:
```diff
@@ -0,0 +1,48 @@
+}
+
+.header__nav-item:not(.header__nav-item_active)::after {
+    position: absolute;
+    margin-left: .25rem;
+    bottom: 4.6rem;
```
Вот из-за `bottom: 4.6rem` у тебя ховер и не видно, так как в основной верстке от отсчитывал отступ снизу от родительского блока, а здесь, в сторибуке, он считает от самого себя расстояние снизу в примерно 74 пикселя, отчего он выходит за рамки этого самого фрейма элемента сторибука, я смог переписать и основную верстку твою и стиль элемента в сторибуке, чтобы и ховер отображался, что в верстке, что в сторибуке корректно. Советую тоже поискать как так можно сделать и переписать что в основной верстке элемент, что в стиле компонента.

Jhon-Crow replied:
в основном файле разве что margin-top: -.8rem; вместо bottom: 4.6rem; написать придумал.
пока просто удалю bottom

Yashiandr commented on src/components/header-nav-item/header-nav-item-component.stories.js:
```diff
@@ -0,0 +1,23 @@
+};
+
+export const Default = Template.bind({});
+Default.args = {
+	variant: 'primary',
+// 	чтоб работали вариации - надо менять темплейст скорее всего
```
чтобы работали вариации, а вернее изменять у компонента его аргументы, надо много чего сделать, начиная от создания в классе компонента переменных аргумента, заканчивая да, темплейт.js и стайл.js каждого компонента, Иван это потом покажет как мы это будем делать, пока не парься и ебашь всё одним видом компонента, но предполагая, что потом ты сможешь изменить стиль компонента уже внутри самого этого компонента.

Yashiandr commented on src/components/button/button-component.styles.js:
```diff
@@ -0,0 +1,49 @@
+		  }
+		  
+		  .button_contact {
+    border: .125rem solid var(--white);
+    color: var(--white);
+}
+
+.button_light {
+    color: var(--darker-violet);
+    background: var(--button-gradient) no-repeat;
+}
+
```
отформатируй

Yashiandr commented on src/components/header-nav-item/header-nav-item-component.styles.js:
```diff
@@ -0,0 +1,48 @@
+
+h1, h2, h3, p {
+    margin: 0;
+    padding: 0;
+}
+
+li {
+    list-style-type: none;
+}
+
+.header__nav-item {
+    /*color: var(--white);*/
+    color: var(--darkest-violet)
+    text-decoration: none;
+    border-top: .1rem solid var(--nav-item-bordr);
+}
+
+.header__nav-item:not(.header__nav-item_active)::after {
+    position: absolute;
+    margin-left: .25rem;
+    bottom: 4.6rem;
+    border-radius: 10rem;
+    background-color: var(--lightest-violet);
+    content: "";
+    height: .5rem;
+    width: .5rem;
+    transform: scale(0);
+    transition: transform .2s ease-in-out;
+}
+
+.header__nav-item_active {
+    border-bottom: .1rem solid var(--nav-item-bordr_active);
+}
+
+.header__nav-item:hover:after {
+    transform: scale(1);
+}
+
+          </style>
+`;
```
отформатируй файл

jhamish committed:
добавил header-nav и header-logo

jhamish committed:
добавил header

jhamish committed:
начал делать hero-section-title, надо будет разобраться со спанами

jhamish committed:
начал делать hero-section-title, надо будет разобраться со спанами

jhamish committed:
закинул из вёрстки апдейты

jhamish committed:
добавил первую картинку

jhamish committed:
сделал первую секцию

jhamish committed:
сделал вторую секцию
надо вынести section-title и section-title_slim отдельно

jhamish committed:
вынести параграф из about-section

jhamish committed:
вынес параграф из about-section

jhamish committed:
добавил service-section-content

jhamish committed:
добавил ещё 2 секции
надо добавить optin

jhamish committed:
добавил optin-section
далее делаю footer

jhamish committed:
начал footer соц.
добавить footer__socialize

jhamish committed:
добавил footer__socialize
добавить footer__line

jhamish committed:
перенёс всё
надо сделать, чтоб правильно отображалось

Yashiandr commented on .storybook/preview.js:
```diff
@@ -0,0 +1,150 @@
+};
+
+
+[ButtonComponent, HeaderNavItemComponent, HeaderNavComponent, HeaderLogoComponent, LineWrapperComponent, HeaderLineWrapperComponent, HeaderComponent, HeroSectionTitleComponent, HeroSectionImageFaceComponent, HeroSectionParagraphComponent, HeroSectionLineWrapperComponent, HeroSectionAddressComponent, HeroSectionAddressItemComponent, HeroSectionComponent, HeroSectionCartComponent, SectionLineCompon ...
```
Отредактируй это не одной строчкой массива, потом фиг найдешь нужный компонент в случае чего
Лучше так:
```
[
  компонент,
  компонент,
  компонент,
].map....
```

Yashiandr commented on src/components/about-section-cart/about-section-cart-component.styles.js:
```diff
@@ -0,0 +1,25 @@
+
+.about-section__paragraph {
+    max-width: 39.37rem;
+}
```
внутри тэга стайл все сдвинуть на таб направо

Yashiandr commented on src/components/about-section-cart/about-section-cart-component.styles.js:
```diff
@@ -0,0 +1,25 @@
+    max-width: 39.37rem;
+}
+
+
```
лишшние пустые строчки убрать

Yashiandr commented on src/components/header-nav-item/header-nav-item-component.template.js:
```diff
@@ -0,0 +1,6 @@
+
+export default `
+    ${styles}
+   <li class='header__nav-item'><a href='#'>Item</a></li>
```
переделай этот компонент как LinkComponent, и сделай из него один <a></a>, тэг же li добавляй на уровне header-nav-component, то бишь там шаблон будет 
```
<ul>
  <li><link-component></link-component></li>
  <li><link-component></link-component></li>
  <li><link-component></link-component></li>
  ...
  <li><link-component></link-component></li>
</ul>
```

Yashiandr commented on src/components/hero-section-address-item/hero-section-address-item-component.template.js:
```diff
@@ -0,0 +1,11 @@
+    ${styles}
+<div class='address-item-location'></div>
+<div>
+<span class='item-title font-weight-700'>Item Span66666666666666666666</span>
```
666666666666666666666666666666666666666

Yashiandr replied:
А вообще вот такие вот моки в виде добавления ненужного текста ради того, чтобы она нормально отверстывалась это недостаток композиции блоков html css

Yashiandr commented on src/components/header-logo/header-logo-component.styles.js:
```diff
@@ -0,0 +1,13 @@
+<style>
+@import url('../common.css');
+
+.header__logo {
```
пропиши этому
display: Inline-block, чтобы у тебя изображения на дефолтной странице этого компонента не было одно слева одно справа, display: block, который по умолчанию просто старается занять все возможное пространство по ширине.

Yashiandr replied:
Но я бы вообще разделил это лого на два изображения

Yashiandr commented on src/components/hero-section-image-face/hero-section-image-face-component.js:
```diff
@@ -0,0 +1,22 @@
+import template from './hero-section-image-face-component.template.js';
+
+export class HeroSectionImageFaceComponent extends HTMLElement {
```
Учти, то все компоненты с вложенными изображениями в бэкграунд потом станут одним компонентом и у тебя из-за этого дофига компонентов, которые выполняют одну и туже функцию

Jhon-Crow replied:
пока так оставлю, если что пере лопачу, как пойму как надо

Jhon-Crow replied:
то есть везде одну картинку поставить, один и тот же компонент в разных секциях?

Yashiandr commented on src/components/optin-section-input-textarea/optin-section-input-textarea-component.styles.js:
```diff
@@ -0,0 +1,27 @@
+export default `
+<style>
+@import url('../common.css');
+
+.optin-section__input_textarea {
+    width: 100%;
+    height: 100%;
+    min-height: 10rem;
+    max-height: 13.125rem;
+    flex-grow: 2;
+}
+
+.optin-section__pulls-border {
+    color: var(--white);
+    border-radius: 2.5rem;
+    border: .125rem solid var(--white);
```
не вставил сюда стиль сброса для Input и получил фигню на выходе

Yashiandr commented on src/components/main/main-component.template.js:
```diff
@@ -0,0 +1,17 @@
+export default `
+    ${styles}
+<main class='main__background main-font body'>
+<section class='main vector-image-for-hero-section-right'>
```
у тебя вот конструкция
```
<mаin clаss="main">
  <sеction clаss="section">
    ...
  </sеction>
</mаin>
```

она зачем такая, ну это как div в dive, лучше все свойства секции, которые нужны в верстке перенеси в main, мейну пропиши также flex-direction: column и всё, и адаптив появится, и лишнего тэга не будет и всем будет хорошо.

Yashiandr commented on src/components/main/main-component.styles.js:
```diff
@@ -0,0 +1,77 @@
+button {
+    border: none;
+}
+
+input {
+    margin: 0;
+    padding: 0;
+    border: none;
+    outline: none;
+    background: none;
+    -webkit-appearance: none;
+    -moz-appearance: none;
+    appearance: none;
+}
+
+textarea {
+    box-sizing: border-box;
+    height: auto;
+    resize: none;
+    border: none;
+    outline: none;
+    background-color: transparent;
+    font-family: inherit;
+    padding: 0;
+    margin: 0;
+}
+
+a {
+    text-decoration: none;
+    color: inherit;
+}
+
+h1, h2, h3, p {
+    margin: 0;
+    padding: 0;
+}
+
+li {
+    list-style-type: none;
+}
```
Компоненты изолированы друг от друга и стили сброса на внутренние тэги не подтягиваются, каждому компоненту необходимо прописывать его собственные стили сброса, либо же, если это необходимо много раз делать, то можно добавить стиль сброса в common.css, но зачастую нет и достаточно указать стиль сброса внутри компонента.

Yashiandr reviewed (commented):
Отформатируй код пж, внутри экспорта файла js всё должно уже на один таб сдвинуто вправо, дальше всё подчиняется уже кодстайлу html и css, то бишь внутри тэга информация на один таб сдвинуто вправо (тоже касается и тэга <style> в style.js файлах).

как пример сравни, что понятнее читать
![image](https://github.com/jegius/ivan-edu/assets/115163286/3884295c-3775-4433-82ab-be46a15c6873)
![image](https://github.com/jegius/ivan-edu/assets/115163286/f564005b-f012-4b9b-a0af-486b74f0d969)

jhamish committed:
+- разобрался со стилями сброса
надо отформатировать

jhamish committed:
исправил адресс и поменял текст для правильного отображения процесс

-отформатировать
-отредачить майн
-переделать линк компонент

jhamish committed:
переделал хедер-линк компонент

-отформатировать
-отредачить майн

jhamish committed:
переделал майн, убрал див в диве

-отформатировать
-почему то странно отображается оптин-сэкшн (не работает флекс гроу и вайдс 100%)

Yashiandr commented on src/components/optin-section-input-big/optin-section-input-big-component.js:
```diff
@@ -0,0 +1,22 @@
+import template from './optin-section-input-big-component.template.js';
```
optin-section-input'ы различны между собой лишь стилями, поэтому они один и тотже компонент, что big, что small, что textarea

Jhon-Crow replied:
а как тут делать? не уловил

Yashiandr commented on src/components/section-line/section-line-component.js:
```diff
@@ -0,0 +1,22 @@
+import template from './section-line-component.template.js';
```
Этот компонент в мейн-компоненте врапается раньше положенного.

Jhon-Crow replied:
можно так оставить? а то очень сжать текст придётся, чтоб не врапало так

Yashiandr commented on src/components/header-nav/header-nav-component.template.js:
```diff
@@ -0,0 +1,12 @@
+
+export default `
+    ${styles}
+<ul class='header__nav'>
```
оберни ul в nav

Yashiandr commented on src/components/header/header-component.template.js:
```diff
@@ -0,0 +1,11 @@
+
+export default `
+    ${styles}
+<ul class='header'>
```
почему это ul а не header?

Yashiandr reviewed (commented):
Отформатируй уже код, без этого всё равно не приму.

jhamish committed:
отформатировал

-почему то странно отображается оптин-сэкшн (не работает флекс гроу и вайдс 100%)

jhamish committed:
сделал что понял

Yashiandr commented on src/components/optin-section/optin-section-component.template.js:
```diff
@@ -0,0 +1,14 @@
+import styles from './optin-section-component.styles.js';
```
![image](https://github.com/jegius/ivan-edu/assets/115163286/fe515825-5c20-4461-b8d9-da2088137898)
Этот компонент должен выглядеть примерно так:
![image](https://github.com/jegius/ivan-edu/assets/115163286/16e0532d-0267-43ac-bd8e-0530e40e9895)

Тебе не надо три компонента одного и того же, в дальнейшем форма инпута будет изменяться атрибутами, которые появятся, когда будешь добавлять адаптивность компонентам

Yashiandr reviewed (commented):
Отформатируй уже код везде, и посмотри в optin-section-component.js, я там оставил комментарий того, что должно там быть.

jhamish committed:
точно отформатировал)

jhamish committed:
начал мудрить с инпутами оптин-сэкш.

jhamish committed:
всё ещё не растягивается инпут на 100% в врапере

jhamish committed:
победил инпуты

-удалить лишнее (pull-wrapper, input-ы биг и тхтэрия)

jhamish committed:
почистил

jhamish committed:
поправил адаптив

jhamish committed:
перенёс кредитс из компонента в майн
поправил по мелочи
-надо исправить перенос в футере

jhamish committed:
поправил мелочи
далее тупик(

jhamish committed:
скопипастил бекграунд в превью .storybook

jhamish committed:
убрал лишние инпуты в оптин

jhamish committed:
поправил сервис-айтемы и мелочи
-сделать белый шрифт в компоненаы
-поменять гапы чтоб убиралось в 1080 по ширине

jhamish committed:
сделал белый шрифт в компоненаы
-поменять гапы чтоб убиралось в 1200 по ширине

jhamish committed:
врапается на 1400 по ширине

jhamish committed:
убрал комменты

Jhon-Crow committed:
Merge branch 'master' into 1-storybook
